### PR TITLE
UIOR-391 Templates prevent the changing PO format

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
- Order templates are preventing the changing of POL order format
- Add `Required` prop to vendor ref number and vendor ref number type in POL Form
- Remove validation for  vendor ref number and vendor ref number type  from template form

https://issues.folio.org/browse/UIOR-391
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/POLFields/VendorFields/FieldRefNumberType.js [hunk 1]
```diff
@@ -18,15 +19,35 @@ const REF_NUMBER_TYPE_OPTIONS = Object.keys(REF_NUMBER_TYPE).map((key) => ({
...
-    dataOptions={REF_NUMBER_TYPE_OPTIONS}
-    fullWidth
-    label={<FormattedMessage id="ui-orders.vendor.refNumberType" />}
-    name="vendorDetail.refNumberType"
-    validate={[requiredRefNumber]}
-  />
-);
+const FieldRefNumberType = ({ dispatch, change, required }) => {
+  const onChange = (_, value) => {
+    dispatch(change('vendorDetail.refNumberType', value));
```
Comment (aliaksei-chumakou) [inline src/common/POLFields/VendorFields/FieldRefNumberType.js]: I still don't get why we need this `onChange` - your field is `vendorDetail.refNumberType` already.

File: src/common/POLFields/VendorFields/FieldVendorRefNumber.js [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
 import { Field } from 'redux-form';
 
 import { TextField } from '@folio/stripes/components';
 
 import { requiredRefNumberType } from '../../../components/Utils/Validate';
 
-const FieldVendorRefNumber = () => {
+const FieldVendorRefNumber = ({ dispatch, change, required }) => {
+  const onChange = (_, value) => {
+    dispatch(change('vendorDetail.refNumber', value));
```
Comment (aliaksei-chumakou) [inline src/common/POLFields/VendorFields/FieldVendorRefNumber.js]: you keep putting dispatches of actions, but it should be working via redux without them explicit

File: src/components/POLine/Vendor/VendorForm.js [hunk 3]
```diff
@@ -45,12 +62,18 @@ const VendorForm = ({ order, accounts }) => {
...
+  dispatch: PropTypes.func.isRequired,
   accounts: PropTypes.arrayOf(PropTypes.object),
   order: PropTypes.object.isRequired,
+  vendorRefNumberType: PropTypes.string,
+  vendorRefNumber: PropTypes.string,
 };
 
 VendorForm.defaultProps = {
   accounts: [],
+  vendorRefNumber: '',
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Vendor/VendorForm.js]: I'm not sure if need these values at all, it's handled by redux form and initialValues

File: src/common/POLFields/VendorFields/FieldRefNumberType.js (hunk 1 above)
Comment (NikitaSedyx) [reply src/common/POLFields/VendorFields/FieldRefNumberType.js]: the same question

Comment (AndreiShumski2) [reply src/common/POLFields/VendorFields/FieldRefNumberType.js]: Fixed

File: src/common/POLFields/VendorFields/FieldVendorRefNumber.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/common/POLFields/VendorFields/FieldVendorRefNumber.js]: Fixed. It works

File: src/settings/OrderTemplates/OrderTemplatesEditor/POLineVendorForm/POLineVendorForm.js [hunk 4]
```diff
@@ -13,21 +14,29 @@ import {
...
+const POLineVendorForm = ({ change, dispatch }) => {
   return (
     <Row>
       <Col
         xs={3}
         data-col-order-template-vendor-number
       >
-        <FieldVendorRefNumber />
+        <FieldVendorRefNumber
+          change={change}
```
Comment (aliaksei-chumakou) [inline src/settings/OrderTemplates/OrderTemplatesEditor/POLineVendorForm/POLineVendorForm.js]: I think you can remove all these passes of `change` and `dispatch`

Comment (AndreiShumski2) [reply src/settings/OrderTemplates/OrderTemplatesEditor/POLineVendorForm/POLineVendorForm.js]: Removed all unnecessary prop

**Pull request merged**